app/code/Magento/Customer/Test/Mftf/Test/StorefrontVerifyCustomCustomerAddressAttributeTest.xml

```diff
@@ -0,0 +1,144 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="StorefrontVerifyCustomCustomerAddressAttributeTest">
+        <annotations>
+            <features value="Customer"/>
+            <stories value="Storefront Custom Customer Address Attribute"/>
+            <title value="Verify Custom Customer Address Attribute Value Shows On Storefront Customer Checkout"/>
+            <description value="Verify that custom customer address attribute value shows at checkout on storefront for second address"/>
+            <severity value="CRITICAL"/>
+            <testCaseId value="MC-37569"/>
+            <group value="customer"/>
+        </annotations>
+        <before>
+            <actionGroup ref="AdminLoginActionGroup" stepKey="loginAsAdmin"/>
+            <createData entity="SimpleProduct2" stepKey="simpleProduct"/>
+            <createData entity="Simple_Customer_Without_Address" stepKey="simpleCustomer"/>
+        </before>
+        <after>
+            <deleteData createDataKey="simpleProduct" stepKey="deleteSimpleProduct"/>
+            <deleteData createDataKey="simpleCustomer" stepKey="deleteCustomer"/>
+        </after>
+
+        <!--Create new custom customer address attribute-->
+        <amOnPage url="{{CustomerAddressAttributesPage.url}}" stepKey="navigateToCustomerAddress"/>
+        <waitForPageLoad time="30" stepKey="timeToLoadAttributes"/>
+        <actionGroup ref="AdminAddOptionsCustomerAttribute" stepKey="adminCreateCustomerFirstAttribute">
+            <argument name="defaultLabel" value="{{AttributeDropdownData.label}}"/>
+            <argument name="attributeCode" value="{{AttributeDropdownData.code}}"/>
+            <argument name="inputType" value="{{AttributeDropdownData.inputType}}"/>
+            <argument name="sortOrder" value="{{AttributeDropdownData.sortOrder}}"/>
+            <argument name="firstOption" value="{{AttributeDropdownData.firstOption}}"/>
+            <argument name="secondOption" value="{{AttributeDropdownData.secondOption}}"/>
+        </actionGroup>
+
+        <!--Add address to B2C Customer-->
+        <actionGroup ref="AdminOpenCustomerEditPageActionGroup" stepKey="openCustomerEditPage">
+            <argument name="customerId" value="$$simpleCustomer.id$"/>
+        </actionGroup>
+        <click selector="{{AdminEditCustomerAddressesSection.addresses}}" stepKey="proceedToAddresses"/>
```

Comment:
would be cleaner & less maintenance to
- change your created customer to `Simple_US_Customer` so you have 1 pre-created address
- change this part of the test to use `OpenEditCustomerAddressFromAdminActionGroup` (this will replace lines 47-58)

doing the above seems like it will give the same coverage as well since the `Add New Address` form is the same as the `Edit Address` form. let me know if i'm wrong and there's a reason to create a new address vs editing an existing one